e2etest/pkg/container/container.go

```diff
@@ -0,0 +1,16 @@
+package container
+
+import "os/exec"
+
+// RunCommand executes the given command / args in a container when the container name
+// is specified, on the current host otherwise.
+func RunCommand(containerName, cmd string, args ...string) (string, error) {
```

Comment:
Would it make sense to put the logic to run a command in the host in a different function? Like
```
func RunHostCommand(cmd string) (string, error) {
                // Didn't test this, but something like this to get the params
                cmd_split := strings.Split(cmd, " ")
                cmd = cmd_split[0]
                args = cmd_split[1:]

                // Or maybe CommandContext and use a timeout?
		out, err := exec.Command(cmd, args).CombinedOutput()
		return string(out), err
}

func RunCommand(containerName, cmd string) (string, error) {
     return RunHostCommand("docker exec" + " " + containerName + " " + "cmd")
}
```

Did it on github so whitespaces and such are horrible, but to show the idea

Having a different function for RunHostCommand and reusing the code seems less hacky than using containerName="" IMHO. Also, we can receive a string and the user doesn't need to split the commands perfectly so maybe it is slightly simpler to use?

What do you think?

Reply:
This did not feel clean from the beginning, but the issue is that we are using these as base for executing commands in the container, wrapped by other calls (you may see it in the part you did not review yet). Moreover, we want to maintain the flexibility of running the same tests in a non dockerized environment.

I can't use an optional global container name either, because at some point we'll need to have multiple frr containers mimicking multiple bgp peers.
In any case, I'll try to rework this to see if I can make it slightly better.

Reply:
Done via creating an `Executor` interface that represent what is gonna execute the actions we want to apply.
Feels less hacky than if-ing on the containerName while preserving the same degree of flexibility.